### ISSUE TITLE
Cannot read property 'restore' of undefined

### ISSUE THREAD

**Issue by tleskin** [simplabs/ember-simple-auth#1822]
Ember: 3.7.1
ESA: 1.6.0
Ember CLI Qunit 4.3.2

We've been upgrading our Ember app in our company from Ember 2.18 -> 3.x, I got it up to 3.7. We were using ESA 1.4.0 but I noticed some issues with tests so I upgraded to 1.5.0 and 1.6.0 to try to get it working. 

I've been trying to start getting one test working then move on to the others and here's the basic test
```
export default function sessionData() {
  return {
    data: {
      attributes: {
        token: 'mock-token',
        refresh: 'mock-refresh'
      },
      type: 'jwt',
      id: ''
    }
  };
}

```

```
import sessionData from 'app/tests/helpers/session-data';
import { authenticateSession } from 'ember-simple-auth/test-support';
import { module, test } from 'qunit';
import { setupApplicationTest } from 'ember-qunit';
import { visit, currentURL, click } from '@ember/test-helpers';
import setupMirage from 'ember-cli-mirage/test-support/setup-mirage';

module('Acceptance | bulk manage', function(hooks){
  setupApplicationTest(hooks);
  setupMirage(hooks);

  hooks.beforeEach(async function() {
    window.localStorage.clear();
    authenticateSession(sessionData());
    this.server.create('company-profile');
  });

  test('visiting /bulk-manage', async function(assert) {
    await visit('/users/bulk-manage');
    assert.equal(currentURL(), '/users/bulk-manage', 'transitions to bulk manage page');
  });
});
```

The issue is that it's running into an error in on _bindToStoreEvents in internal-session.js.

https://github.com/simplabs/ember-simple-auth/blob/0ab8847ed94812c391f66f79c645127abd771f95/addon/internal-session.js#L189-L190

Basically line 189 `const authenticator = this._lookupAuthenticator(authenticatorFactory);` is returning undefined and it's trying to call restore on the next line on undefined. The authenticatorFactory is "authenticator:test". 

I'm not sure if this is a bug or something is missing from our setup.

**Comment by marcoow** [comment]
If you're using ember-cli-qunit 4.1.0 or earlier, you have to pass the application instance to the `authenticateSession` helper - see https://github.com/simplabs/ember-simple-auth#ember-cli-qunit-410-and-earlier. Could that be it?

**Comment by tleskin** [comment]
@marcoow It's on ember-cli-qunit 4.3.2.

**Comment by Turbo87** [comment]
@tleskin without a reproduction repository this will be very hard to fix. can you reproduce this in a fresh `ember new` app?

**Comment by antonbavykin1991** [comment]
for me, when I run test, issue happens here https://github.com/simplabs/ember-simple-auth/blob/master/packages/ember-simple-auth/addon/initializers/setup-session-restoration.js#L16 - It cannot find `session:main` in test mode,

**Comment by sdebarros** [comment]
Closing this due to inactivity. Feel free to re-open if it's still an issue.
